extra/cache-extra/Node/CacheNode.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+/*
+ * This file is part of Twig.
+ *
+ * (c) Fabien Potencier
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Twig\Extra\Cache\Node;
+
+use Twig\Compiler;
+use Twig\Node\Expression\AbstractExpression;
+use Twig\Node\Node;
+
+class CacheNode extends Node
+{
+    public function __construct(AbstractExpression $key, ?AbstractExpression $ttl, ?AbstractExpression $tags, Node $body, int $lineno, string $tag)
+    {
+        $nodes = ['key' => $key, 'body' => $body];
+        if (null !== $ttl) {
+            $nodes['ttl'] = $ttl;
+        }
+        if (null !== $tags) {
+            $nodes['tags'] = $tags;
+        }
+
+        parent::__construct($nodes, [], $lineno, $tag);
+    }
+
+    public function compile(Compiler $compiler): void
+    {
+        $compiler
+            ->addDebugInfo($this)
+            ->write('$cached = $this->env->getRuntime(\'Twig\Extra\Cache\CacheRuntime\')->getCache()->get(')
+            ->subcompile($this->getNode('key'))
+            ->raw(", function (\Symfony\Contracts\Cache\ItemInterface \$item) use (\$context) {\n")
+            ->indent()
+            ->write("try {\n")
+            ->indent()
+            ->write("\$item->tag('twig');\n")
```

Comment:
If we remove this, we can also remove the try/catch, which then means that ppl that use `tag()` with a non-taggaware pool will get an exception, which could be nice for DX. That'd be my preference I think.